pyccel/codegen/printing/ccode.py

```diff
@@ -461,10 +461,14 @@ def _print_LiteralComplex(self, expr):
     def _print_PythonComplex(self, expr):
         self._additional_imports.add("complex")
         if expr.is_cast:
-            return self._print(expr.internal_var)
+            value = self._print(expr.internal_var)
+            type_name = self.find_in_dtype_registry('complex', expr.precision)
+            return '({0})({1})'.format(type_name, value)
         else:
-            return self._print(PyccelAssociativeParenthesis(PyccelAdd(expr.real,
+            value = self._print(PyccelAssociativeParenthesis(PyccelAdd(expr.real,
                             PyccelMul(expr.imag, LiteralImaginaryUnit()))))
+            type_name = self.find_in_dtype_registry('complex', expr.precision)
+            return '({0})({1})'.format(type_name, value)
```

Comment:
These lines seem to be repeated could you not use the if block to determine `value` then have these 2 lines in common?